### PR TITLE
Add preliminary nix files

### PR DESCRIPTION
**Pull request by tfc**
Hi there,

i love this example code and just out of interest i made it build with nix. Maybe it is helpful to others who would like to play with it, too.

### CODE REVIEW

File: release.nix [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    '';
+    installPhase = ''
+      mkdir -p $out/bin
+      cp ${name} $out/bin/
+    '';
+  };
+in {
+  inherit temptory;
+  treap = f "treap";
+  treap_manual = f "treap_manual";
```
Comment (deech) [inline release.nix:22]: Just FYI, `treap` also needs `-lgc` since it's not relying on manual memory management.

**Comment by deech** [comment]
I'll merge this but a followup PR that fixes the `-lgc` issue would be nice. My knowledge of Nix is basically 0.

**Comment by tfc** [comment]
> I'll merge this but a followup PR that fixes the `-lgc` issue would be nice. My knowledge of Nix is basically 0.

Oh, that's of course not a problem to supply on the command line, too, but i wonder why this then even builds...

**Comment by tfc** [comment]
@deech where does the `gc` lib come from? i am a bit riddled with why treap.dats compiles without `-lgc` at all and when i explicitly add `-lgc` then `ld` tells me it doesn't find it.

**Comment by deech** [comment]
[`libgc`](https://linux.die.net/man/1/libgc) is an add-on garbage collector, it's weird that it compiles and runs without that. Arch [seems to package it](https://archlinux.org/packages/extra/x86_64/gc/files/) so I'm sure NixPkgs has some equivalent.

**Pull request merged**